assets/scripts/build_gallery.py

```diff
@@ -219,6 +248,7 @@ def build_gallery(
     Build the gallery object by running diagnostics for each Expectation and returning the resulting reports.
 
     Args:
+        only_combine: if true, only combine the various backend_full.json files
```

Comment:
Could we refactor the combining out into a separate method instead of using a param here?

Reply:
Would be nice, but in order to do the combine, it has to iterate over every Expectation to get all that metadata (and the in-memory objects for the Expectation instance and diagnostics object). The Expectation instance and some attributes of the diagnostic object need to get passed to a static method to recalculate the maturity checklist and final maturity status.

Since it has to do this one final run, we do sqllite here (so sqlite wouldn't be included in the parametrized calls that will get pushed later).

When a CI stage writes whatever _full.json file, it could have written some metadata.json file that got uploaded (and re-uploaded) to S3, but those in-memory objects are needed anyway, so no point in a metadata.json

Reply:
I'm not sure I'm following here - why couldn't the final merge be a separate function like `merge_expectation_metadata_for_gallery()`? `build_gallery()` already has too many responsibilities.

Reply:
In the refactor, this [combine_backend_results](https://github.com/great-expectations/great_expectations/blob/misc/gallery-split-join-logic/assets/scripts/build_gallery.py#L280) func was added, but it needs to be passed in ExpectationDiagnostic objects and instances of the Expectations so the maturity_checklist, library_metadata, and coverage_score can be recalculated. The files being combined don't have a ton of info in them (snippets below).

Also see this comment https://github.com/great-expectations/great_expectations/pull/7572#discussion_r1172437866

https://superconductive-public.s3.us-east-2.amazonaws.com/static/gallery/pandas_full.json

```
{
    "expect_batch_row_count_to_match_prophet_date_model": {
        "backend_test_result_counts": [
            {
                "backend": "pandas",
                "num_passed": 2,
                "num_failed": 0,
                "failing_names": null
            }
        ]
    },
    "expect_column_average_lat_lon_pairwise_distance_to_be_less_than": {
        "backend_test_result_counts": [
            {
                "backend": "pandas",
                "num_passed": 4,
                "num_failed": 0,
                "failing_names": null
            }
        ]
    },
    "expect_column_average_to_be_within_range_of_given_point": {
        "backend_test_result_counts": [
            {
                "backend": "pandas",
                "num_passed": 2,
                "num_failed": 0,
                "failing_names": null
            }
        ]
    },
    ...
}
```

https://superconductive-public.s3.us-east-2.amazonaws.com/static/gallery/spark_full.json

```
{
    "expect_column_discrete_entropy_to_be_between": {
        "backend_test_result_counts": [
            {
                "backend": "spark",
                "num_passed": 4,
                "num_failed": 0,
                "failing_names": null
            }
        ]
    },
    "expect_column_distinct_values_to_be_in_set": {
        "backend_test_result_counts": [
            {
                "backend": "spark",
                "num_passed": 10,
                "num_failed": 0,
                "failing_names": null
            }
        ]
    },
    "expect_column_distinct_values_to_contain_set": {
        "backend_test_result_counts": [
            {
                "backend": "spark",
                "num_passed": 15,
                "num_failed": 0,
                "failing_names": null
            }
        ]
    },
    ...
}
```

Reply:
Ok - that makes sense, thank you for clarifying. Could gathering the ExpectationDiagnostic objects and instances of the Expectations be called also though in a `merge_expectation_metadata_for_gallery()` function? There already is a `get_expectation_instances()` function.

Reply:
get_expectation_instances is the only thing that gets the instances and it's called one time. What is wrong with passing in the result to combine_backend_results when the time is right?